### PR TITLE
Improve UPPMAX configs

### PR DESCRIPTION
**Pull request by MaxUlysse**
EDIT:
- replace `-profile uppmax_devel` by `-profile uppmax,devel`
- automatic assignation of resources based on hostnames
- update docs

Let's simplify some configs.
So let's start by getting ridd of `uppmax_devel`:

```Bash
≻nextflow run . -profile uppmax
[...]
Config Profile    : uppmax
Config Description: Swedish UPPMAX cluster profile provided by nf-core/configs.
Config Contact    : Phil Ewels (@ewels)
Config URL        : https://www.uppmax.uu.se/
```

```Bash
≻nextflow run . -profile uppmax,devel
[...]
Profile    : uppmax,devel
Config Description: Testing & development profile for UPPMAX, provided by nf-core/configs.
Config Contact    : Phil Ewels (@ewels)
Config URL        : https://www.uppmax.uu.se/
```

### CODE REVIEW

**Comment by apeltzer** [comment]
Love it - did anyone test this already?

**Comment by MaxUlysse** [comment]
I'm beginning a test on UPPMAX right now, but first test on my computer looked good, as you can see in the first post ;-)
params were changed

File: conf/uppmax.config [hunk 1]
```diff
@@ -7,6 +7,7 @@ params {
 
 singularity {
   enabled = true
+  cacheDir = '/sw/data/uppnex/ToolBox/nf-core'
```
Comment (ewels) [inline conf/uppmax.config]: Will this break stuff if people don't have write permissions and the image is not already there?

Comment (MaxUlysse) [reply conf/uppmax.config]: The plan is to have all nf-core pipelines singularity images there whenever we have releases at some point

**Comment by MaxUlysse** [comment]
@apeltzer Tests are good:

```
[maxime@rackham3 ~]$ nextflow run nf-core/sarek -r 2.5.1 -profile uppmax,devel --project XXX --custom_config_base https://raw.githubusercontent.com/MaxUlysse/nf-core_configs/UPPMAX --input https://github.com/nf-core/test-datasets/raw/sarek/testdata/tsv/tiny-manta-https.tsv --genome smallGRCh37 --igenomesIgnore --genomes_base https://github.com/nf-core/test-datasets/raw/sarek/reference
N E X T F L O W  ~  version 19.10.0
Launching `nf-core/sarek` [deadly_gutenberg] - revision: 05c6e0b632 [2.5.1]
[2m----------------------------------------------------
                                        ,--./,-.
        ___     __   __   __   ___     /,-._.--~'
  |\ | |__  __ /  ` /  \ |__) |__         }  {
  | \| |       \__, \__/ |  \ |___     \`-._,-`-,
                                        `._,._,'
    ____
  .´ _  `.
 /  |\`-_ \      __        __   ___     
|   | \  `-|    |__`  /\  |__) |__  |__/
 \ |   \  /     .__| /¯¯\ |  \ |___ |  \
  `|____\´

  nf-core/sarek v2.5.1
----------------------------------------------------
Pipeline Release  : 2.5.1
Run Name          : deadly_gutenberg
Max Resources     : 120 GB memory, 16 cpus, 1h time per job
Container         : singularity - nfcore/sarek:2.5.1
Input             : https://github.com/nf-core/test-datasets/raw/sarek/testdata/tsv/tiny-manta-https.tsv
Step              : mapping
Save Genome Index : No
Nucleotides/s     : 1000.0
Output dir        : ./results
Launch dir        : /domus/h1/maxime
Working dir       : /domus/h1/maxime/work
Script dir        : /home/maxime/.nextflow/assets/nf-core/sarek
User              : maxime
genome            : smallGRCh37
fasta             : https://github.com/nf-core/test-datasets/raw/sarek/reference/human_g1k_v37_decoy.small.fasta
intervals         : https://github.com/nf-core/test-datasets/raw/sarek/reference/small.intervals
dbsnp             : https://github.com/nf-core/test-datasets/raw/sarek/reference/dbsnp_138.b37.small.vcf.gz
knownIndels       : [https://github.com/nf-core/test-datasets/raw/sarek/reference/1000G_phase1.indels.b37.small.vcf.gz, https://github.com/nf-core/test-datasets/raw/sarek/reference/Mills_and_1000G_gold_standard.indels.b37.small.vcf.gz]
Config Profile    : uppmax,devel
Config Description: Testing & development profile for UPPMAX, provided by nf-core/configs.
Config Contact    : Phil Ewels (@ewels)
Config URL        : https://www.uppmax.uu.se/
```

```
[maxime@rackham3 ~]$ nextflow run nf-core/sarek -r 2.5.1 -profile uppmax --project XXX --custom_config_base https://raw.githubusercontent.com/MaxUlysse/nf-core_configs/UPPMAX --input https://github.com/nf-core/test-datasets/raw/sarek/testdata/tsv/tiny-manta-https.tsv --genome smallGRCh37 --igenomesIgnore --genomes_base https://github.com/nf-core/test-datasets/raw/sarek/reference
N E X T F L O W  ~  version 19.10.0
Launching `nf-core/sarek` [jolly_galileo] - revision: 05c6e0b632 [2.5.1]
[2m----------------------------------------------------
                                        ,--./,-.
        ___     __   __   __   ___     /,-._.--~'
  |\ | |__  __ /  ` /  \ |__) |__         }  {
  | \| |       \__, \__/ |  \ |___     \`-._,-`-,
                                        `._,._,'
    ____
  .´ _  `.
 /  |\`-_ \      __        __   ___     
|   | \  `-|    |__`  /\  |__) |__  |__/
 \ |   \  /     .__| /¯¯\ |  \ |___ |  \
  `|____\´

  nf-core/sarek v2.5.1
----------------------------------------------------
Pipeline Release  : 2.5.1
Run Name          : jolly_galileo
Max Resources     : 128 GB memory, 16 cpus, 10d time per job
Container         : singularity - nfcore/sarek:2.5.1
Input             : https://github.com/nf-core/test-datasets/raw/sarek/testdata/tsv/tiny-manta-https.tsv
Step              : mapping
Save Genome Index : No
Nucleotides/s     : 1000.0
Output dir        : ./results
Launch dir        : /domus/h1/maxime
Working dir       : /domus/h1/maxime/work
Script dir        : /home/maxime/.nextflow/assets/nf-core/sarek
User              : maxime
genome            : smallGRCh37
fasta             : https://github.com/nf-core/test-datasets/raw/sarek/reference/human_g1k_v37_decoy.small.fasta
intervals         : https://github.com/nf-core/test-datasets/raw/sarek/reference/small.intervals
dbsnp             : https://github.com/nf-core/test-datasets/raw/sarek/reference/dbsnp_138.b37.small.vcf.gz
knownIndels       : [https://github.com/nf-core/test-datasets/raw/sarek/reference/1000G_phase1.indels.b37.small.vcf.gz, https://github.com/nf-core/test-datasets/raw/sarek/reference/Mills_and_1000G_gold_standard.indels.b37.small.vcf.gz]
Config Profile    : uppmax
Config Description: Swedish UPPMAX cluster profile provided by nf-core/configs.
Config Contact    : Phil Ewels (@ewels)
Config URL        : https://www.uppmax.uu.se/
```

**Comment by MaxUlysse** [comment]
Now adding automatic cpus/memory requirements based on hostnames:

```Bash
[maxime@rackham3 ~]$ nextflow run nf-core/sarek -r 2.5.1 -profile uppmax --project XXX --custom_config_base https://raw.githubusercontent.com/MaxUlysse/nf-core_configs/UPPMAX --input https://github.com/nf-core/test-datasets/raw/sarek/testdata/tsv/tiny-manta-https.tsv --genome smallGRCh37 --igenomesIgnore --genomes_base https://github.com/nf-core/test-datasets/raw/sarek/reference
N E X T F L O W  ~  version 19.10.0
Launching `nf-core/sarek` [gloomy_poincare] - revision: 05c6e0b632 [2.5.1]
[2m----------------------------------------------------
                                        ,--./,-.
        ___     __   __   __   ___     /,-._.--~'
  |\ | |__  __ /  ` /  \ |__) |__         }  {
  | \| |       \__, \__/ |  \ |___     \`-._,-`-,
                                        `._,._,'
    ____
  .´ _  `.
 /  |\`-_ \      __        __   ___     
|   | \  `-|    |__`  /\  |__) |__  |__/
 \ |   \  /     .__| /¯¯\ |  \ |___ |  \
  `|____\´

  nf-core/sarek v2.5.1
----------------------------------------------------
Pipeline Release  : 2.5.1
Run Name          : gloomy_poincare
Max Resources     : 125 GB memory, 20 cpus, 10d time per job
Container         : singularity - nfcore/sarek:2.5.1
Input             : https://github.com/nf-core/test-datasets/raw/sarek/testdata/tsv/tiny-manta-https.tsv
Step              : mapping
Save Genome Index : No
Nucleotides/s     : 1000.0
Output dir        : ./results
Launch dir        : /domus/h1/maxime
Working dir       : /domus/h1/maxime/work
Script dir        : /home/maxime/.nextflow/assets/nf-core/sarek
User              : maxime
genome            : smallGRCh37
fasta             : https://github.com/nf-core/test-datasets/raw/sarek/reference/human_g1k_v37_decoy.small.fasta
intervals         : https://github.com/nf-core/test-datasets/raw/sarek/reference/small.intervals
dbsnp             : https://github.com/nf-core/test-datasets/raw/sarek/reference/dbsnp_138.b37.small.vcf.gz
knownIndels       : [https://github.com/nf-core/test-datasets/raw/sarek/reference/1000G_phase1.indels.b37.small.vcf.gz, https://github.com/nf-core/test-datasets/raw/sarek/reference/Mills_and_1000G_gold_standard.indels.b37.small.vcf.gz]
Config Profile    : uppmax
Config Description: Swedish UPPMAX cluster profile provided by nf-core/configs.
Config Contact    : Phil Ewels (@ewels)
Config URL        : https://www.uppmax.uu.se/
```

**Comment by MaxUlysse** [comment]
@alneberg @ewels Can you check UPPMAX requirements?

**Comment by alneberg** [comment]
Need to use a regex for hostname detection I think. For example on an interactive session on irma:

```
[alneberg@i167 ~]$ hostname
i167.uppmax.uu.se
```

**Comment by MaxUlysse** [comment]
regex for node/cluster works:
```Bash
[maxime@rackham1 ~]$ nextflow run nf-core/sarek -r 2.5.1 -profile uppmax --project XXX --custom_config_base https://raw.githubusercontent.com/MaxUlysse/nf-core_configs/UPPMAX --input https://github.com/nf-core/test-datasets/raw/sarek/testdata/tsv/tiny-manta-https.tsv --genome smallGRCh37 --igenomesIgnore --genomes_base https://github.com/nf-core/test-datasets/raw/sarek/reference
N E X T F L O W  ~  version 19.10.0
Launching `nf-core/sarek` [modest_einstein] - revision: 05c6e0b632 [2.5.1]
[2m----------------------------------------------------
                                        ,--./,-.
        ___     __   __   __   ___     /,-._.--~'
  |\ | |__  __ /  ` /  \ |__) |__         }  {
  | \| |       \__, \__/ |  \ |___     \`-._,-`-,
                                        `._,._,'
    ____
  .´ _  `.
 /  |\`-_ \      __        __   ___     
|   | \  `-|    |__`  /\  |__) |__  |__/
 \ |   \  /     .__| /¯¯\ |  \ |___ |  \
  `|____\´

  nf-core/sarek v2.5.1
----------------------------------------------------
Pipeline Release  : 2.5.1
Run Name          : modest_einstein
Max Resources     : 125 GB memory, 20 cpus, 10d time per job
Container         : singularity - nfcore/sarek:2.5.1
Input             : https://github.com/nf-core/test-datasets/raw/sarek/testdata/tsv/tiny-manta-https.tsv
Step              : mapping
Save Genome Index : No
Nucleotides/s     : 1000.0
Output dir        : ./results
Launch dir        : /domus/h1/maxime
Working dir       : /domus/h1/maxime/work
Script dir        : /home/maxime/.nextflow/assets/nf-core/sarek
User              : maxime
genome            : smallGRCh37
fasta             : https://github.com/nf-core/test-datasets/raw/sarek/reference/human_g1k_v37_decoy.small.fasta
intervals         : https://github.com/nf-core/test-datasets/raw/sarek/reference/small.intervals
dbsnp             : https://github.com/nf-core/test-datasets/raw/sarek/reference/dbsnp_138.b37.small.vcf.gz
knownIndels       : [https://github.com/nf-core/test-datasets/raw/sarek/reference/1000G_phase1.indels.b37.small.vcf.gz, https://github.com/nf-core/test-datasets/raw/sarek/reference/Mills_and_1000G_gold_standard.indels.b37.small.vcf.gz]
Config Profile    : uppmax
Config Description: Swedish UPPMAX cluster profile provided by nf-core/configs.
Config Contact    : Phil Ewels (@ewels)
Config URL        : https://www.uppmax.uu.se/
```

File: conf/uppmax.config (hunk 1 above)
Comment (MaxUlysse) [inline conf/uppmax.config]:
```suggestion
```
But you're right, as it's not working now, I'll remove it from this PR, and it'll come in a new one

File: conf/uppmax.config [hunk 2]
```diff
@@ -17,9 +17,36 @@ process {
...
+profiles {
+  devel {
+    params {
+      config_profile_description = 'Testing & development profile for UPPMAX, provided by nf-core/configs.'
+      // Max resources to be requested by a devel job
+      max_memory = 120.GB
+      max_time = 1.h
+    }
+    executor.queueSize = 1
+    process.queue = 'devel'
```
Comment (ewels) [inline conf/uppmax.config]: Is the devel queue available on all clusters?

File: docs/uppmax.md [hunk 3]
```diff
@@ -50,13 +40,27 @@ Note that each job will still start with the same request as normal, but restart
 
 All jobs will be submitted to fat nodes using this method, so it's only for use in extreme circumstances.
 
-## Uppmax-devel config
+## How to specify a UPPMAX cluster
+
+You actually do not need to, based on `hostnames`, configuration will be automatically applied for the cluster you are on following these specifications:
```
Comment (ewels) [inline docs/uppmax.md]:
```suggestion
The UPPMAX nf-core configuration profile uses the `hostname` of the active environment to automatically apply the following resource limits:
```

File: conf/uppmax.config (hunk 2 above)
Comment (MaxUlysse) [reply conf/uppmax.config]: We had the `uppmax_devel` profile before, so I'm guessing the one responsible for that did check it out ;-)

**Pull request merged**